src/textual_serve/download_manager.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass, field
+import logging
+from typing import AsyncGenerator, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from textual_serve.app_service import AppService
+
+log = logging.getLogger("textual-serve")
+
+DOWNLOAD_TIMEOUT = 4
+DOWNLOAD_CHUNK_SIZE = 1024 * 64  # 64 KB
+
+
+@dataclass
+class Download:
+    app_service: "AppService"
+    delivery_key: str
+    file_name: str
+    open_method: str
+    mime_type: str
+    encoding: str | None = None
+    """The encoding of the content.
+    Will be None if the content is binary.
+    """
+    incoming_chunks: asyncio.Queue[bytes | None] = field(default_factory=asyncio.Queue)
```

Comment:
We don't buffer anything in the queue, there will never be more than 1 packet in it. It's just a channel through which packets can be delivered to the aiohttp handler.

Reply:
It could be an `Event` if there is a single item. But this does leave us with the potential of reading packets in advance to better saturate the bandwidth. So it might be best leaving it as a queue for now.